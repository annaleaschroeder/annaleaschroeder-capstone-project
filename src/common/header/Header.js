import React from 'react'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

Header.propTypes = {
  date: PropTypes.object.isRequired,
}

const month = [
  'January',
  'February',
  'March',
  'April',
  'May',
  'June',
  'July',
  'August',
  'September',
  'October',
  'November',
  'December',
]

export default function Header({ date, title }) {
  const year = date.getFullYear()
  const monthNumber = date.getMonth()
  const newMonth = month[monthNumber]

  return (
    <HeaderStyled>
      <Headline>{title}</Headline>
      <SubHeadline>{newMonth + ' ' + year}</SubHeadline>
    </HeaderStyled>
  )
}

const HeaderStyled = styled.header`
  width: 100%;
  z-index: 200;
  background: var(--blue-main);
  background-size: cover;
  padding: 10px;
  text-align: center;
`

const Headline = styled.h1`
  text-align: center;
  margin: 0;
  font-size: 200%;
  display: block;
  color: var(--white);
`
const SubHeadline = styled.h2`
  text-align: center;
  margin: 0;
  font-size: 150%;
  display: block;
  color: var(--white);
`
