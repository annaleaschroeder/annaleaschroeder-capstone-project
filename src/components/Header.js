import React from 'react'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

Header.propTypes = {
  date: PropTypes.object.isRequired,
}

export default function Header({ date }) {
  const month = [
    'January',
    'February',
    'March',
    'April',
    'May',
    'June',
    'July',
    'August',
    'September',
    'October',
    'November',
    'December',
  ]

  const year = date.getFullYear()
  const monthNumber = date.getMonth()
  const newMonth = month[monthNumber]

  return (
    <HeaderStyled>
      <Headline>Budget Overview</Headline>
      <SubHeadline>{newMonth + ' ' + year}</SubHeadline>
    </HeaderStyled>
  )
}

const HeaderStyled = styled.header`
  position: fixed;
  width: 100vw;
  z-index: 200;
  background-color: hotpink;
  padding: 10px;
`

const Headline = styled.h1`
  text-align: center;
  margin: 0%;
  font-size: 200%;
`
const SubHeadline = styled.h2`
  text-align: center;
  margin: 0%;
  font-size: 150%;
`
