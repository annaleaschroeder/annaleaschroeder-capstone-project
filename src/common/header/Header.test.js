import React from 'react'
import { render } from '@testing-library/react'
import Header from './Header'
import renderer from 'react-test-renderer'

describe('Header', () => {
  const date = new Date()
  const title = 'Budget Overview'
  it('shows the main headline', () => {
    const { getByText } = render(<Header date={date} title={title} />)
    expect(getByText('Budget Overview')).toBeInTheDocument()
  })

  it('renders month and year into the subheadline with date object', () => {
    const month = [
      'January',
      'February',
      'March',
      'April',
      'May',
      'June',
      'July',
      'August',
      'September',
      'October',
      'November',
      'December',
    ]

    const { getByText } = render(<Header date={date} />)
    expect(
      getByText(`${month[date.getMonth()]} ${date.getFullYear()}`)
    ).toBeInTheDocument()
  })

  it('renders month and year into the subheadline mock object', () => {
    const { getByText } = render(
      <Header date={{ getFullYear: () => 2020, getMonth: () => 7 }} />
    )
    expect(getByText('August 2020')).toBeInTheDocument()
  })

  it('renders correctly for January', () => {
    const tree = renderer.create(
      <Header date={new Date('2020-01-07 19:29:59 +02:00')} />
    )
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly for February', () => {
    const tree = renderer.create(
      <Header date={new Date('2020-02-27 19:29:59 +02:00')} />
    )
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly for December', () => {
    const tree = renderer.create(
      <Header date={new Date('2020-12-31 09:15:10 +02:00')} />
    )
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly for January 2021', () => {
    const tree = renderer.create(
      <Header date={new Date('2021-01-15 00:00:00 +02:00')} />
    )
    expect(tree).toMatchSnapshot()
  })
})
