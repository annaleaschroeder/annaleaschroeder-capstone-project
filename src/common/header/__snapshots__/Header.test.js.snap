// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly for December 1`] = `
<header
  className="Header__HeaderStyled-sc-1hwuuy3-0 dbgkfT"
>
  <h1
    className="Header__Headline-sc-1hwuuy3-1 cMMJAe"
  />
  <h2
    className="Header__SubHeadline-sc-1hwuuy3-2 loWqPP"
  >
    December 2020
  </h2>
</header>
`;

exports[`Header renders correctly for February 1`] = `
<header
  className="Header__HeaderStyled-sc-1hwuuy3-0 dbgkfT"
>
  <h1
    className="Header__Headline-sc-1hwuuy3-1 cMMJAe"
  />
  <h2
    className="Header__SubHeadline-sc-1hwuuy3-2 loWqPP"
  >
    February 2020
  </h2>
</header>
`;

exports[`Header renders correctly for January 1`] = `
<header
  className="Header__HeaderStyled-sc-1hwuuy3-0 dbgkfT"
>
  <h1
    className="Header__Headline-sc-1hwuuy3-1 cMMJAe"
  />
  <h2
    className="Header__SubHeadline-sc-1hwuuy3-2 loWqPP"
  >
    January 2020
  </h2>
</header>
`;

exports[`Header renders correctly for January 2021 1`] = `
<header
  className="Header__HeaderStyled-sc-1hwuuy3-0 dbgkfT"
>
  <h1
    className="Header__Headline-sc-1hwuuy3-1 cMMJAe"
  />
  <h2
    className="Header__SubHeadline-sc-1hwuuy3-2 loWqPP"
  >
    January 2021
  </h2>
</header>
`;
